pkg/azuredisk/controllerserver.go

```diff
@@ -435,7 +436,61 @@ func (d *Driver) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (
 
 // ControllerExpandVolume controller expand volume
 func (d *Driver) ControllerExpandVolume(ctx context.Context, req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "ControllerExpandVolume is not yet implemented")
+	if len(req.GetVolumeId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID missing in request")
+	}
+	if err := d.ValidateControllerServiceRequest(csi.ControllerServiceCapability_RPC_EXPAND_VOLUME); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid expand volume request: %v", req)
+	}
+
+	capacityBytes := req.GetCapacityRange().GetRequiredBytes()
+	if capacityBytes == 0 {
+		return nil, status.Error(codes.InvalidArgument, "volume capacity range missing in request")
+	}
+	requestSize := *resource.NewQuantity(capacityBytes, resource.BinarySI)
+
+	diskURI := req.GetVolumeId()
+	if err := isValidDiskURI(diskURI); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "disk URI(%s) is not valid: %v", diskURI, err)
+	}
+	if !isManagedDisk(diskURI) {
+		return nil, status.Errorf(codes.InvalidArgument, "the disk type(%s) is not ManagedDisk", diskURI)
+	}
+
+	diskName, err := getDiskName(diskURI)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not get disk name from diskURI(%s) with error(%v)", diskURI, err)
+	}
+	resourceGroup, err := getResourceGroupFromURI(diskURI)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not get resource group from diskURI(%s) with error(%v)", diskURI, err)
+	}
+
+	disk, err := d.cloud.DisksClient.Get(ctx, resourceGroup, diskName)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not get the disk(%s) under rg(%s) with error(%v)", diskName, resourceGroup, err)
+	}
+	if disk.DiskProperties.DiskSizeGB == nil {
+		return nil, status.Errorf(codes.Internal, "could not get size of the disk(%s)", diskName)
+	}
+	oldSize := *resource.NewQuantity(int64(*disk.DiskProperties.DiskSizeGB), resource.BinarySI)
+
+	klog.V(2).Infof("expanding azure disk(%s)", diskURI)
```

Comment:
`begin to expand azure disk(%s) with new size(%v)`

Reply:
done.